src/click/core.py

```diff
@@ -1661,7 +1661,7 @@ def handle_parse_result(self, ctx, opts, args):
         with augment_usage_errors(ctx, param=self):
             value = self.consume_value(ctx, opts)
             try:
-                value = self.full_process_value(ctx, value)
+                value = self.full_process_value(ctx, value, is_flag=self.name in opts)
```

Comment:
Options already have an `is_flag` attribute. Can we take advantage of that? This probably also ties in with #549.

Reply:
An idea I have is to set `is_flag=True` for prompt options that have `prompt_required=False`